x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/jobs_list_status.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { DATAFEED_STATE, JOB_STATE } from '../../../../../../ml/common';
+import { MLManageJobsLink } from '../../../shared/Links/MachineLearningLinks/MLManageJobsLink';
+
+export function JobsListStatus({
+  jobId,
+  jobState,
+  datafeedState,
+  version,
+}: {
+  jobId: string;
+  jobState?: JOB_STATE;
+  datafeedState?: DATAFEED_STATE;
+  version: number;
+}) {
+  const jobIsOk =
+    jobState === JOB_STATE.OPENED || jobState === JOB_STATE.OPENING;
+
+  const datafeedIsOk =
+    datafeedState === DATAFEED_STATE.STARTED ||
+    datafeedState === DATAFEED_STATE.STARTING;
+
+  const isClosed =
+    jobState === JOB_STATE.CLOSED || jobState === JOB_STATE.CLOSING;
+
+  const isLegacy = version < 3;
+
+  const statuses: React.ReactElement[] = [];
+
+  if (jobIsOk && datafeedIsOk) {
+    statuses.push(
+      <EuiBadge color="success">
+        {i18n.translate(
+          'xpack.apm.settings.anomalyDetection.jobList.okStatusLabel',
+          { defaultMessage: 'OK' }
+        )}
+      </EuiBadge>
+    );
+  } else if (!isClosed) {
```

Comment:
nit: flat `if`'s are a bit easier to parse when there are many of them

```suggestion
  }
  
  if (!isClosed) {
```